### PR TITLE
inject context for batching loops

### PR DESCRIPTION
**Pull request by aniketmaurya**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

</details>

## What does this PR do?

Batching loops currently lacks `context` injection. This PR enables that.

- [x] updated tests
- [x] updated README

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/litserve/server.py [hunk 1]
```diff
@@ -96,12 +96,24 @@ def run_batched_loop(lit_api, lit_spec, request_queue: Queue, request_buffer, ma
...
+                _inject_context(
+                    context,
+                    lit_api.decode_request,
+                    input,
+                )
+                for input, context in zip(inputs, contexts)
+            ]
             x = lit_api.batch(x)
-            y = lit_api.predict(x)
+            y = _inject_context(contexts, lit_api.predict, x)
```
Comment (aniketmaurya) [inline src/litserve/server.py:113]: `predict` will receive a list of `context` which is right but feels weird.

**Review by lantiga — COMMENTED** [review]
Looks good, each request needs to have its own context, so a list of them is the only sensible thing.
Please add to README and we can proceed!

**Comment by aniketmaurya** [comment]
hi @lantiga, I have added the README changes. could you take a look again please?

**Review by lantiga — APPROVED** [review]
Good to go!

**Pull request merged**